src/components/ContextFeedTimeline.jsx

```diff
@@ -0,0 +1,319 @@
+                style: { userSelect: "none" },
+              }}
+            />
+          ) : null,
+        )}
+        {/* Selection Area (shaded background) shown once time window selection made */}
+        {selectedMinMillis && selectedMaxMillis ? (
+          <ReferenceArea
+            x1={selectedMinMillis}
+            x2={selectedMaxMillis}
+            stroke="none"
+            fillOpacity={0.2}
+          />
+        ) : null}
+        {/* Data Points & Lines */}
+        {Object.values(SAL_INDEX_INFO)
+          .filter(
+            (info) =>
+              info.displayIndex != null &&
+              info.label !== "AUTOLOG" &&
+              info.label !== "Narrative Log",
+          )
+          .map((info) => {
+            const displayIndex = info.displayIndex;
+            const label = info.label;
+            const color = info.color;
+            const isActive =
+              activeLabels.length === 0 || activeLabels.includes(label);
+            const opacity = isActive ? 1 : 0.1;
+
+            return (
+              <Fragment key={displayIndex}>
+                {/* Horizontal lines behind data points */}
+                <ReferenceLine
+                  // Take from 10 to display top to bottom
+                  y={10 - displayIndex}
+                  stroke="#606060"
+                  strokeOpacity={opacity}
+                  strokeWidth={1}
+                  ifOverflow="visible"
+                />
+                {/* Data points */}
+                <Line
+                  data={data
+                    .filter((d) => d.displayIndex === displayIndex)
+                    .map((d) => ({ ...d, y: 10 - displayIndex }))}
+                  dataKey="y"
+                  stroke=""
+                  dot={(props) => {
+                    const { key } = props;
+                    return (
+                      <CustomisedDot
+                        key={key}
+                        cx={props.cx}
+                        cy={props.cy}
+                        stroke={color}
+                        opacity={opacity}
+                        h="16"
+                        w="1"
+                      />
+                    );
+                  }}
+                  isAnimationActive={false}
+                />
+              </Fragment>
+            );
+          })}
+        {/* Selection area (rectangle outline) shown once time window selection made */}
+        {selectedMinMillis && selectedMaxMillis ? (
+          <ReferenceArea
+            x1={selectedMinMillis}
+            x2={selectedMaxMillis}
+            stroke="hotPink"
+            fillOpacity={0}
+          />
+        ) : null}
+        {/* Selection rectangle shown during active highlighting */}
+        {refAreaLeft && refAreaRight ? (
+          <ReferenceArea
+            x1={refAreaLeft}
+            x2={refAreaRight}
+            fillOpacity={0.3}
+            fill="pink"
+          />
+        ) : null}
```

Comment:
```suggestion
        {refAreaLeft && refAreaRight && (
          <ReferenceArea
            x1={refAreaLeft}
            x2={refAreaRight}
            fillOpacity={0.3}
            fill="pink"
          />
        )}
```
I was going to suggest the same [logical AND conditional rendering approach](https://react.dev/learn/conditional-rendering#logical-and-operator-), but I stepped back because `selectedMinMillis && selectedMaxMillis` seem to be numeric so could lead to the pitfall explained on the docs.

Reply:
`selectedMinMillis`/`selectedMaxMillis` and `refAreaLeft`/`refAreaRight` are both numeric, but I don't see how they can ever be `0`. They're either some large integer like `1753259328477` or they're `null`. Buuuut, just in case, I've added the `> 0` checks like the docs suggest.